src/server/pps/server/api_server.go

```diff
@@ -1954,8 +1967,18 @@ func RepoNameToEnvString(repoName string) string {
 	return strings.ToUpper(repoName)
 }
 
-// Convert a persist.JobInfo into a Kubernetes batch.Job spec
-func job(kubeClient *kube.Client, jobInfo *persist.JobInfo, jobShimImage string, jobImagePullPolicy string) (*batch.Job, error) {
+type jobOptions struct {
+	labels             map[string]string
+	parallelism        int32
+	userImage          string
+	jobShimImage       string
+	jobImagePullPolicy string
+	jobEnv             []api.EnvVar
+	volumes            []api.Volume
+	volumeMounts       []api.VolumeMount
+}
+
+func getJobOptions(kubeClient *kube.Client, jobInfo *persist.JobInfo, jobShimImage string, jobImagePullPolicy string) (*jobOptions, error) {
```

Comment:
Ahh ofc, in that case you could call it `newJobOptions` since it's a constructor... but considering this method isn't even exported publicly I think you should just ignore this, `getJobOptions` seems fine, it gets the point across.